drivers/input/touchscreen/tsc2007_core.c

```diff
@@ -141,9 +142,8 @@ static irqreturn_t tsc2007_soft_irq(int irq, void *handle)
 
 			rt = ts->max_rt - rt;
 
+			touchscreen_report_pos(ts->input, &ts->prop, tc.x, tc.y, false);
```

Comment:
You've changed the order of this (which triggers ABS_X and ABS_Y events), wrt to BTN_TOUCH. Does that matter?

I suspect not as I believe `input_sync` is the trigger to do anything with all the events, but I couldn't say so definitively.

Reply:
yes they will only be interpreted after the sync call so order of setting the report should be irrelevant.
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation/input/input-programming.rst?id=HEAD#n121